src/main/io/osd.c

```diff
@@ -518,17 +518,23 @@ static bool osdDrawSingleElement(uint8_t item)
         }
 
     case OSD_CRAFT_NAME:
+        // This does not strictly support iterative updating if the craft name changes at run time. But since the craft name is not supposed to be changing this should not matter, and blanking the entire length of the craft name string on update will make it impossible to configure elements to be displayed on the right hand side of the craft name.
+        //TODO: When iterative updating is implemented, change this so the craft name is only printed once whenever the OSD 'flight' screen is entered.
+
         if (strlen(pilotConfig()->name) == 0) {
             strcpy(buff, "CRAFT_NAME");
         } else {
-            for (unsigned int i = 0; i < MAX_NAME_LENGTH; i++) {
-                buff[i] = toupper((unsigned char)pilotConfig()->name[i]);
-                if (pilotConfig()->name[i] == 0) {
-                    buff[i] = SYM_BLANK;
-                }
-            }
-            buff[MAX_NAME_LENGTH] = '\0';
+            unsigned i;
+            for (i = 0; i < MAX_NAME_LENGTH; i++) {
+                if (pilotConfig()->name[i]) {
+                    buff[i] = toupper((unsigned char)pilotConfig()->name[i]);
+                } else {
+                    break;
+                }    
+            }    
+            buff[i] = '\0';
```

Comment:
As of my prev comment on the old version of this PR. Please do `buff[i] = SYM_BLANK;` instead. Or even nothing at all. I do not think there is a need for a terminated string in the screen buffer. 
And the risk for garbage after the string might still be a problem.

Reply:
See my previous comment, and the discussion in #5203.

Reply:
@AndersHoglund : `displayWrite()` expects null-terminated string ...

Reply:
Ah, OK. I stand corrected.

Reply:
What about the risk for trailing garbage? When is this screen area cleared? At power up init?
What happens for example when changing from a long to a short name?

Reply:
See comment in this pull request.